projects/step-core/src/lib/modules/tree/services/tree-state.service.ts

```diff
@@ -137,16 +137,34 @@ export class TreeStateService<T, N extends TreeNode> implements OnDestroy {
     this.editNodeIdInternal$.next(undefined);
   }
 
-  toggleSkip(): void {
-    const nodeId = this.selectedNodeIds$.value[0];
-    const node = this.findNodeById(nodeId);
+  toggleSkip(actionId?: string): void {
+    const nodeIds = this.selectedNodeIds$.value;
+    const nodes: any[] = [];
 
-    if (!node) {
+    if (!nodeIds.length) {
       return;
+    } else {
+      nodeIds.forEach(id => {
+        nodes.push(this.findNodeById(id));
+      })
     }
 
-    const isSkipped = !node.isSkipped;
-    const isUpdated = this._treeNodeUtils.updateNodeData(this.originalRoot!, nodeId, { isSkipped });
+    let isUpdated = false;
+
+    nodes.forEach(node => {
+      let isSkipped;
+      if (actionId) {
+        isSkipped = actionId === PlanTreeAction.DISABLE ? true : false;
```

Comment:
this doesn't work fine: isUpdated = isUpdated || this._treeNodeUtils.updateNodeData(this.originalRoot!, node.id, { isSkipped });

changed it to: isUpdated = this._treeNodeUtils.updateNodeData(this.originalRoot!, node.id, { isSkipped }) || isUpdated

Reply:
Actually const isSkipped = forceSkip || !node.isSkipped; also doesn't work like it should, with that code if we choose enable for enabled nodes, then it will disabled them, which is very misleading for user.